### PR TITLE
Fix Makefile to now build temp files when not temp

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@kolyshkin I could not get the foreach to work, in the limited time I played with it.  But this should fix the leaked tmpfiles.

File: Makefile [hunk 1]
```diff
@@ -43,13 +43,13 @@ ifndef GOPATH
...
-noselinux:=$(shell mktemp /tmp/noselinux_XXXX)
-selinux:=$(shell mktemp /tmp/selinux_XXXX)
 test: check-gopath
-	grep '^func [A-Z].*' go-selinux/selinux_stub.go > ${noselinux}
-	grep '^func [A-Z].*' go-selinux/selinux_linux.go > ${selinux}
-	diff ${noselinux} ${selinux}
-	rm -f ${noselinux} ${selinux}
+	$(eval NOSELINUX := $(shell mktemp /tmp/noselinux_XXXX))
+	$(eval SELINUX := $(shell mktemp /tmp/selinux_XXXX))
+	grep '^func [A-Z].*' go-selinux/selinux_stub.go > ${NOSELINUX};
```
Comment (kolyshkin) [inline Makefile:49]: It seems that `;` at EOL is redundant here (and in the lines below).

The alternative to is something like
```
diff <(grep '^func [A-Z].*' go-selinux/selinux_stub.go) <(grep ^func [A-Z].*' go-selinux/selinux_linux.go)
```
but it is bash/zsh specific

Comment (rhatdan) [reply Makefile:49]: That is a nice elegant solution.

Comment (rhatdan) [reply Makefile:49]: But getting it working in Makefile, will be difficult.

Comment (kolyshkin) [reply Makefile:49]: why?
```make
checkme:
        bash -c "diff  <(grep '^func [A-Z].*' go-selinux/selinux_stub.go) <(grep '^func [A-Z].*' go-selinux/selinux_linux.go)"
```

```console
$ make  checkme
[kir@kir-rhat selinux]$ make checkme 
bash -c "diff  <(grep '^func [A-Z].*' go-selinux/selinux_stub.go) <(grep '^func [A-Z].*' go-selinux/selinux_linux.go)"
13a14
> func SetTaskLabel(label string) error {
make: *** [Makefile:35: checkme] Error 1
```

(i have deliberately removed `SetTaskLabel` for demo purposes)

Comment (kolyshkin) [reply Makefile:49]: Just noticed that `.*` in grep regex is redundant

Comment (rhatdan) [reply Makefile]: Because I am an idiot and was trying to do this in diff command rather then launching bash.

**Comment by rhatdan** [comment]
I don't know where that comes from.

**Comment by kolyshkin** [comment]
> ^ looks like pull approve needs updating; is that a configuration file somewhere?

I think I had it updated (there's that sync button on pullapprove dot com site) and it started to show my name in "needs 2 approval from ..." message, but apparently there's some more sorcery involved. @caniszczyk maybe you can help?

**Pull request merged**